### PR TITLE
Add Git tutorial

### PR DESCRIPTION
**Pull request by Starlight220**
Articles:
- [x] Introduction
- [x] Create Branch
- [x] Add and Commit - ~~missing 1 images~~
- [x] Push
- [x] Submit PR
- [x] Reviews
- [x] Rebasing - ~~missing 1 images~~
- [x] Squash/Merge 

Other TODOs: 
- ~~Handle SVG linking~~
- [x] Add menu, vertical sidebar
- [x] Easier access from main menu
- [x] link to [k88hudson/git-flight-rules](https://github.com/k88hudson/git-flight-rules/blob/master/README.md)

### CODE REVIEW

**Comment by NirLevanon** [comment]
Who is this for? Definitely not for beginners :)
Very well built and documented, but the question if it fits to be used by the other sub-team members and those to come in the next years, meaning - can they handle this level of English and terminology?

Again, very well documented! 👏 

What are the odds someone (other than you) can translate this to Hebrew, as a side-project? This will be very helpful to other teams as well.

File: tutorials/git/commit.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+permalink: /tutorials/git/commit.html
+layout: tutorial
+---
+Commits can be thought of as points in the history of the repository, or groups of changes.
+Each commit should be given a meaningful name that sums up the changes in that commit.
+Please avoid situations like this:
```
Comment (NirLevanon) [inline tutorials/git/commit.md:7]: Add emphasis (bold or color)

File: tutorials/git/commit.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+```ps
+git add file1 file2 file3
+```
+The `*` wildcard can be used.
+
+To create a commit, use `git commit`:
+```ps
+git commit -m "message"
+```
+
```
Comment (NirLevanon) [inline tutorials/git/commit.md:31]: If possible, try to figure out how to ident the inner section, for readbility

Comment (NirLevanon) [reply tutorials/git/commit.md:31]: This pertains to **ALL** usage of the details component.

File: tutorials/git/create-branch.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
+---
+permalink: /tutorials/git/create-branch.html
+layout: tutorial
+---
+Creating a new branch allows you to add commits on a parallel version of the repository.
+This can be used to test new features and not affect the `main` branch.
+Later, the changes made in the branch can be added to the `main` branch once they are tested.
+
```
Comment (NirLevanon) [inline tutorials/git/create-branch.md:8]: Why not add a visual example of how branching looks like, meaning a branching graph. A clear one, where a branching is shown. Don't think showing merge is relevant.

Comment (NirLevanon) [reply tutorials/git/create-branch.md:8]: Okay, you can copy the image from the index.

**Review by NirLevanon — APPROVED** [review]

**Comment by Starlight220** [comment]
I plan to have a whole `tutorials` section on the website, with tutorials for other subteams too. 
The Markdown files are automagically converted to HTML and formatted using CSS, together with some JS magic. 

You can run the site locally by cloning, [installing Jekyll](https://jekyllrb.com/docs/installation/), and running `bundle exec jekyll serve` in the project root. See how they're rendered, if there's still formatting changes that you think should be done then tell me. 

I totally agree with the need to add some more graphs/diagrams. 

About the level of the tutorial, I tried to make it as simple as possible. I don't want to translate it to Hebrew because it loses all the terminology. If you think of anything that can make it more level-appropriate for both current and future subteam members, I'm happy to implement it. 

I have an idea:
@korozya-bit - can you go over the _content_ (text, not the HTML elements etc) of the tutorial and see if the level is too high? Mostly the `.md` files - the content is there. You can also run the website on your computer using the steps I wrote above to see the rendered pages with the formatting and images and everything.

**Comment by Starlight220** [comment]
IntelliJ 2020.2.3 (released today) changed the Git interface, so if the changes are significant I will redo the IntelliJ sections. This might delay when this is merged. 

I might split the core tutorial "framework" stuff to a different PR so it can be merged earlier.

**Pull request merged**